### PR TITLE
Move hybridoverlay master into the master controller

### PR DESCRIPTION
**Pull request by JacobTanenbaum**
moving the master side hybrid overlay components into the master controller makes the code easier to understand and removes spurious messages from the logs.                                                                                                                                                                
                                                                                                                                                                                                                                                                                                                            
In addition moving towards a hybrid overlay that can be toggled on a running cluster moving this code will make the required allocation of ip addresses easier                                                                                                                                                              
                                                                                                                                                                                                                                                                                                                            
The node components do not make sense to move as there are different code capabilities that we need if the code is running on a windows or a linux node and there are no plans to allow a master running on a windows node.                                                                                                 
                                                                                                                                                                                                                                                                                                                            
                                                                                                            


<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/ovn-org/ovn-kubernetes/blob/master/CONTRIBUTING.m



** Make sure all your commits include a signature generated with `git commit -s` **

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Trivial changes are exempt from following this template.
If your change is non-trivial, please provide the following information:
-->

**- What this PR does and why is it needed**
<!--
A summary of the changes within this pull request and some context
as to why they were made
-->

**- Special notes for reviewers**
<!--
What exactly did you change - you may also defer to information
contained in commit messages. At a bare minimum it's worth highlighting
which areas of the code were changed as it's easier to assign reviewers
-->


**- How to verify it**
<!--
Did you include unit tests? or end-to-end tests?
How can I manually verify that this patch achieves its objective
-->


**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->

### CODE REVIEW

File: go-controller/pkg/ovn/hybrid.go [hunk 1]
```diff
@@ -1,151 +1,55 @@
...
+		_ = oc.hybridOverlaySubnetAllocator.ReleaseNetwork(hostsubnets[0])
 		return nil, err
 	}
 
 	klog.Infof("Allocated hybrid overlay HostSubnet %s for node %s", hostsubnets[0], node.Name)
 	return hostsubnets[0], nil
 }
 
-func (m *MasterController) releaseNodeSubnet(nodeName string, subnet *net.IPNet) error {
+func (oc *Controller) releaseNodeSubnet(nodeName string, subnet *net.IPNet) error {
```
Comment (dcbw) [inline go-controller/pkg/ovn/hybrid.go:46]: @JacobTanenbaum can we rename this to `releaseHybridOverlaySubnet()` ?

File: go-controller/pkg/ovn/hybrid.go [hunk 2]
```diff
@@ -159,7 +63,7 @@ func (m *MasterController) releaseNodeSubnet(nodeName string, subnet *net.IPNet)
 //   - annotation, no lsp: configure lsp
 //   - annotation, lsp: ensure lsp matches annotation
 //   - no annotation, lsp: set annotation from lsp
-func (m *MasterController) handleOverlayPort(node *kapi.Node, annotator kube.Annotator) error {
+func (oc *Controller) handleOverlayPort(node *kapi.Node, annotator kube.Annotator) error {
```
Comment (dcbw) [inline go-controller/pkg/ovn/hybrid.go:66]: @JacobTanenbaum while we're at it, can we make this `handleHybridOverlayPort()`?

File: go-controller/pkg/ovn/hybrid.go [hunk 3]
```diff
@@ -259,65 +163,17 @@ func (m *MasterController) handleOverlayPort(node *kapi.Node, annotator kube.Ann
 	return nil
 }
 
-func (m *MasterController) deleteOverlayPort(node *kapi.Node) {
+func (oc *Controller) deleteOverlayPort(node *kapi.Node) {
```
Comment (dcbw) [inline go-controller/pkg/ovn/hybrid.go:166]: @JacobTanenbaum let's make this `deleteHybridOverlayPort()`.

File: go-controller/pkg/ovn/master.go [hunk 4]
```diff
@@ -1205,9 +1203,24 @@ func (oc *Controller) syncNodesRetriable(nodes []interface{}) error {
...
+				klog.Warning(err.Error())
+			} else if hostSubnet != nil {
+				klog.V(5).Infof("Node %s contains subnets: %v", node.Name, hostSubnet)
+				if err := oc.hybridOverlaySubnetAllocator.MarkAllocatedNetwork(hostSubnet); err != nil {
+					utilruntime.HandleError(err)
+				}
+			}
+			// there is nothing left to be done if this is a hybrid overlay node
+			continue
+
```
Comment (dcbw) [inline go-controller/pkg/ovn/master.go:1220]: @JacobTanenbaum remove the added newline?

Comment (JacobTanenbaum) [reply go-controller/pkg/ovn/master.go]: done

File: go-controller/pkg/ovn/hybrid.go (hunk 3 above)
Comment (JacobTanenbaum) [reply go-controller/pkg/ovn/hybrid.go]: done

File: go-controller/pkg/ovn/hybrid.go (hunk 2 above)
Comment (JacobTanenbaum) [reply go-controller/pkg/ovn/hybrid.go]: done

File: go-controller/pkg/ovn/hybrid.go (hunk 1 above)
Comment (JacobTanenbaum) [reply go-controller/pkg/ovn/hybrid.go]: @dcbw I renamed it to `relaseHybridOverlayNodeSubnet()` I thought keeping the word Node in the name made it clearer that the function removes the subnet for the node. does that make sense? does that change meet the spirit of your comment or do you prefer `releaseHybridOverlaySubnet()` much better?

Comment (dcbw) [reply go-controller/pkg/ovn/hybrid.go]: Sounds fine to me.

**Review by trozet — APPROVED** [review]

**Pull request merged**